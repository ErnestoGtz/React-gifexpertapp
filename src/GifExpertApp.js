import React,{ Fragment,useState } from 'react';
import { CategoryAdd } from './components/CategoryAdd';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) =>{

    //const categories = ['One Punch','Samurai X','Dragon Ball'];
    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //     //setCategories([...categories,'Yugi-Oh']);
    //     setCategories(categories => [...categories,'Yugi-Oh']);
    // }




    return (
        <Fragment>
            <h2>GifExpertApp</h2>
            <CategoryAdd setCategories={setCategories}/>
            <hr />

            <ol>
                {
                    categories.map( category => 
                        <GifGrid 
                            key={category}
                            category={category}
                        />
                    )
                }
            </ol>
        </Fragment>
    );
}

export default GifExpertApp;