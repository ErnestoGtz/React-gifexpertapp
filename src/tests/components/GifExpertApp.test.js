import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../../GifExpertApp';
import '@testing-library/jest-dom';


describe('Realizando Pruebas <GifExpertApp />', () => {
    test('debe mostrar el componente correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['Ranma','Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
    
});
