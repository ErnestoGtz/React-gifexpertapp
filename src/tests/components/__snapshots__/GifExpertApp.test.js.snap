// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Realizando Pruebas <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Ranma"
      key="Ranma"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Realizando Pruebas <GifExpertApp /> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <CategoryAdd
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
